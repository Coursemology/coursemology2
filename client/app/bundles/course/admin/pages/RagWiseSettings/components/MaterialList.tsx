import { FC, memo } from 'react';
import { List, Typography } from '@mui/material';
import equal from 'fast-deep-equal';
import { Folder } from 'types/course/admin/ragWise';

import Section from 'lib/components/core/layouts/Section';
import useTranslation from 'lib/hooks/useTranslation';

import translations from '../translations';

import ExpandAllSwitch from './buttons/ExpandAllSwitch';
import FolderTab from './FolderTab';

interface MaterialListProps {
  rootFolder: Folder;
}

const MaterialList: FC<MaterialListProps> = (props) => {
  const { rootFolder } = props;
  const { t } = useTranslation();

  return (
    <Section
      contentClassName="flex flex-col space-y-3"
      sticksToNavbar
      subtitle={t(translations.materialsSectionSubtitle)}
      title={t(translations.materialsSectionTitle)}
    >
      <section>
        <div className="flex justify-between items-center mb-4">
          <div>
            <ExpandAllSwitch />
          </div>
          <div className="pr-5 space-x-48 flex justify-end">
            <div className="text-center">
              <Typography
                align="center"
                className="max-w-[10px] pr-24"
                variant="body2"
              >
                {t(translations.knowledgeBaseStatusSettings)}
              </Typography>
            </div>
          </div>
        </div>
        <div>
          <List
            className="p-0 w-full border border-solid border-neutral-300 rounded-lg"
            dense
          >
            <FolderTab folder={rootFolder} level={0} />
          </List>
        </div>
      </section>
    </Section>
  );
};

export default memo(MaterialList, equal);
