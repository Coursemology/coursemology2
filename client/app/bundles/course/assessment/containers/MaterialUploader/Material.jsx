import { Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, defineMessages } from 'react-intl';
import { formatLongDateTime } from 'lib/moment';
import {
  Avatar,
  CircularProgress,
  IconButton,
  ListItem,
  ListItemAvatar,
  ListItemText,
  ListItemSecondaryAction,
} from '@mui/material';
import Assignment from '@mui/icons-material/Assignment';
import Delete from '@mui/icons-material/Delete';
import ReactTooltip from 'react-tooltip';

const styles = {
  avatar: {
    height: '32px',
    width: '32px',
  },
  iconButton: {
    color: 'black',
    width: 24,
    height: 24,
    padding: 4,
    marginRight: 16,
  },
  secondaryText: {
    fontSize: 12,
  },
};

const translations = defineMessages({
  uploading: {
    id: 'course.material.uploading',
    defaultMessage: 'Uploading',
  },
  disableDelete: {
    id: 'course.material.disableDelete',
    defaultMessage:
      'This action is unavailable as the Materials Component is disabled in the Admin Settings',
  },
});

const propTypes = {
  id: PropTypes.number,
  name: PropTypes.string.isRequired,
  updatedAt: PropTypes.string,
  onMaterialDelete: PropTypes.func,
  deleting: PropTypes.bool,
  uploading: PropTypes.bool,
  disabled: PropTypes.bool,
};

class Material extends Component {
  onDelete = (e) => {
    e.preventDefault();
    const { id, name, onMaterialDelete } = this.props;
    if (onMaterialDelete) onMaterialDelete(id, name);
  };

  renderIcon() {
    const { disabled } = this.props;
    if (this.props.deleting || this.props.uploading) {
      return <CircularProgress size={24} style={styles.iconButton} />;
    }

    return (
      <IconButton
        disabled={disabled}
        onClick={this.onDelete}
        style={styles.iconButton}
      >
        <Delete
          data-tip
          data-for="delete-file-button"
          data-tip-disable={!disabled}
        />
        <ReactTooltip id="delete-file-button">
          <FormattedMessage {...translations.disableDelete} />
        </ReactTooltip>
      </IconButton>
    );
  }

  renderSecondaryText() {
    if (this.props.uploading) {
      return <FormattedMessage {...translations.uploading} />;
    }
    const { updatedAt } = this.props;
    return (
      <div style={styles.secondaryText}>{formatLongDateTime(updatedAt)}</div>
    );
  }

  render() {
    const { name } = this.props;

    return (
      <ListItem button>
        <ListItemAvatar>
          <Avatar style={styles.avatar}>
            <Assignment />
          </Avatar>
        </ListItemAvatar>
        <ListItemText primary={name} secondary={this.renderSecondaryText()} />
        <ListItemSecondaryAction>{this.renderIcon()}</ListItemSecondaryAction>
      </ListItem>
    );
  }
}

Material.propTypes = propTypes;

export default Material;
