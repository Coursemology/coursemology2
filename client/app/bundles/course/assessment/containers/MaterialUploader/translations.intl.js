import { defineMessages } from 'react-intl';

const translations = defineMessages({
  deleteSuccess: {
    id: 'course.assessment.materialList.deleteSuccess',
    defaultMessage: '"{name}" was deleted.',
  },
  deleteFail: {
    id: 'course.assessment.materialList.deleteFail',
    defaultMessage: 'Failed to delete "{name}", please try again.',
  },
  uploadFail: {
    id: 'course.assessment.materialList.uploadFail',
    defaultMessage: 'Failed to upload materials.',
  },
});

export default translations;
