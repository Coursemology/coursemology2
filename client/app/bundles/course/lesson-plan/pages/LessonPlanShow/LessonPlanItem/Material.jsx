import PropTypes from 'prop-types';
import Description from '@material-ui/icons/Description';
import { grey } from '@mui/material/colors';

const styles = {
  material: {
    position: 'relative',
    paddingTop: 10,
    paddingLeft: 35,
  },
  icon: {
    height: 20,
    width: 20,
    display: 'block',
    position: 'absolute',
    top: 0,
    left: 0,
    margin: 10,
  },
};

const Material = (props) => {
  const { name, url } = props;
  return (
    <div style={styles.material}>
      <Description style={styles.icon} htmlColor={grey[700]} />
      <a href={url}>{name}</a>
    </div>
  );
};

Material.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default Material;
