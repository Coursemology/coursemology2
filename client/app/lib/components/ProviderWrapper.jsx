import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/lib/integration/react';
import { IntlProvider, addLocaleData } from 'react-intl';
import { i18nLocale } from 'lib/helpers/server-context';
import { MuiThemeProvider, createTheme } from '@material-ui/core/styles';
import {
  createTheme as createThemeV5,
  adaptV4Theme,
  ThemeProvider,
} from '@mui/material/styles';
import LoadingIndicator from 'lib/components/LoadingIndicator';
import zh from 'react-intl/locale-data/zh';
import palette from '../../theme/palette';
import { black, grey, white } from '../../theme/colors';

import ErrorBoundary from './ErrorBoundary';
import translations from '../../../build/locales/locales.json';

const propTypes = {
  store: PropTypes.shape({
    subscribe: PropTypes.func.isRequired,
    dispatch: PropTypes.func.isRequired,
    getState: PropTypes.func.isRequired,
  }),
  persistor: PropTypes.object,
  children: PropTypes.element.isRequired,
};

const themeSettings = {
  palette,
  // https://material-ui.com/customization/themes/#typography---html-font-size
  // https://material-ui.com/style/typography/#migration-to-typography-v2
  typography: {
    htmlFontSize: 10,
  },
  overrides: {
    MuiButton: {
      contained: {
        color: black,
        backgroundColor: white,
      },
    },
    MuiDialogContent: {
      root: {
        color: 'black',
        fontSize: '16px',
        fontFamily: `'Roboto', 'sans-serif'`,
      },
    },
    MuiAccordionSummary: {
      content: {
        margin: 0,
        '&$expanded': { margin: 0 },
      },
    },
    MuiMenuItem: {
      root: {
        height: '48px',
      },
    },
    MuiModal: {
      root: {
        zIndex: 1800,
      },
    },
    MuiStepLabel: {
      iconContainer: {
        paddingLeft: '2px',
        paddingRight: '2px',
      },
    },
    MuiTableCell: {
      root: {
        padding: '8px 14px',
        height: '48px',
      },
      head: {
        color: grey[500],
        padding: '16px 16px',
      },
    },
    MuiTabs: {
      root: {
        backgroundColor: palette.primary.main,
        color: 'white',
      },
    },
  },
};

const theme = createTheme(themeSettings);
const themeV5 = createThemeV5(adaptV4Theme(themeSettings));

const ProviderWrapper = ({ store, persistor, children }) => {
  const availableForeignLocales = { zh };
  const localeWithoutRegionCode = i18nLocale.toLowerCase().split(/[_-]+/)[0];

  let messages;
  if (
    localeWithoutRegionCode !== 'en' &&
    availableForeignLocales[localeWithoutRegionCode]
  ) {
    addLocaleData(availableForeignLocales[localeWithoutRegionCode]);
    messages =
      translations[localeWithoutRegionCode] || translations[i18nLocale];
  }

  let providers = children;

  if (store && persistor) {
    providers = (
      <PersistGate loading={<LoadingIndicator />} persistor={persistor}>
        {providers}
      </PersistGate>
    );
  }

  providers = (
    <IntlProvider locale={i18nLocale} messages={messages}>
      <ThemeProvider theme={themeV5}>
        <MuiThemeProvider theme={theme}>{providers}</MuiThemeProvider>{' '}
      </ThemeProvider>
    </IntlProvider>
  );

  if (store) {
    providers = <Provider store={store}>{providers}</Provider>;
  }

  return <ErrorBoundary>{providers}</ErrorBoundary>;
};

ProviderWrapper.propTypes = propTypes;

export default ProviderWrapper;
