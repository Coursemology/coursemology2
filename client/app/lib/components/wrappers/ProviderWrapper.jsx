import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { ToastContainer } from 'react-toastify';
import { injectStyle } from 'react-toastify/dist/inject-style';
import {
  createTheme,
  StyledEngineProvider,
  ThemeProvider,
} from '@mui/material/styles';
import PropTypes from 'prop-types';
import { PersistGate } from 'redux-persist/lib/integration/react';
import resolveConfig from 'tailwindcss/resolveConfig';
import { grey } from 'theme/colors';
import palette from 'theme/palette';

import LoadingIndicator from 'lib/components/core/LoadingIndicator';
import { i18nLocale } from 'lib/helpers/server-context';

import translations from '../../../../build/locales/locales.json';
import tailwindUserConfig from '../../../../tailwind.config';

import ErrorBoundary from './ErrorBoundary';

injectStyle();

const propTypes = {
  store: PropTypes.shape({
    subscribe: PropTypes.func.isRequired,
    dispatch: PropTypes.func.isRequired,
    getState: PropTypes.func.isRequired,
  }),
  persistor: PropTypes.object,
  children: PropTypes.node.isRequired,
};

const tailwindConfig = resolveConfig(tailwindUserConfig);

const pxInInt = (pixels) => parseInt(pixels.replace('px', ''), 10);

const ProviderWrapper = ({ store, persistor, children }) => {
  const localeWithoutRegionCode = i18nLocale.toLowerCase().split(/[_-]+/)[0];

  // TODO: Replace with React's createRoot once true SPA is ready
  const rootElement = document.getElementById('root');

  const theme = createTheme({
    palette,
    // https://material-ui.com/customization/themes/#typography---html-font-size
    // https://material-ui.com/style/typography/#migration-to-typography-v2
    typography: {
      htmlFontSize: 10,
    },
    breakpoints: {
      values: {
        xs: 0,
        sm: pxInInt(tailwindConfig.theme.screens.sm),
        md: pxInInt(tailwindConfig.theme.screens.md),
        lg: pxInInt(tailwindConfig.theme.screens.lg),
        xl: pxInInt(tailwindConfig.theme.screens.xl),
      },
    },
    components: {
      MuiDialog: {
        defaultProps: { container: rootElement },
        styleOverrides: { root: { zIndex: 100 } },
      },
      MuiPopover: {
        defaultProps: { container: rootElement },
        styleOverrides: { root: { zIndex: 100 } },
      },
      MuiPopper: {
        defaultProps: { container: rootElement },
        styleOverrides: { root: { zIndex: 100 } },
      },
      MuiCard: { styleOverrides: { root: { overflow: 'visible' } } },
      MuiMenuItem: { styleOverrides: { root: { height: '48px' } } },
      MuiModal: { styleOverrides: { root: { zIndex: 100 } } },
      MuiAppBar: {
        styleOverrides: {
          // When there is a MUI Dialog component, somehow
          // the color of appbar is changed... smh
          root: {
            background: `${palette.primary.main} !important`,
            color: 'white !important',
          },
        },
      },
      MuiDialogContent: {
        styleOverrides: {
          root: {
            color: 'black',
            fontSize: '16px',
            fontFamily: `'Roboto', 'sans-serif'`,
          },
        },
      },
      MuiAccordionSummary: {
        styleOverrides: {
          root: { width: '100%' },
          content: {
            margin: 0,
            paddingLeft: '16px',
            '&$expanded': { margin: 0 },
          },
        },
      },
      MuiStepLabel: {
        styleOverrides: {
          iconContainer: {
            paddingLeft: '2px',
            paddingRight: '2px',
          },
        },
      },
      MuiTableCell: {
        styleOverrides: {
          root: {
            padding: '8px 14px',
            height: '48px',
          },
          head: {
            color: grey[500],
            padding: '16px 16px',
          },
          sizeSmall: {
            padding: '6px 12px',
            height: '40px',
          },
        },
      },
    },
  });

  let messages;
  if (localeWithoutRegionCode !== 'en') {
    messages =
      translations[localeWithoutRegionCode] || translations[i18nLocale];
  }

  let providers = (
    <>
      {children}
      <ToastContainer position="bottom-center" />
    </>
  );

  if (store && persistor) {
    providers = (
      <PersistGate loading={<LoadingIndicator />} persistor={persistor}>
        {providers}
      </PersistGate>
    );
  }

  providers = (
    <IntlProvider locale={i18nLocale} messages={messages} textComponent="span">
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={theme}>{providers}</ThemeProvider>
      </StyledEngineProvider>
    </IntlProvider>
  );

  if (store) {
    providers = <Provider store={store}>{providers}</Provider>;
  }

  return <ErrorBoundary>{providers}</ErrorBoundary>;
};

ProviderWrapper.propTypes = propTypes;

export const StoreProviderWrapper = ({ store, persistor, children }) => {
  let providers = children;

  if (store && persistor) {
    providers = (
      <PersistGate loading={<LoadingIndicator />} persistor={persistor}>
        {providers}
      </PersistGate>
    );
  }

  if (store) providers = <Provider store={store}>{providers}</Provider>;

  return providers;
};

StoreProviderWrapper.propTypes = propTypes;

export default ProviderWrapper;
