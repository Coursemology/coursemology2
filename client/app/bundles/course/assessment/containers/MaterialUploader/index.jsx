import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import CourseAPI from 'api/course';
import typeMaterial from 'lib/types';
import MaterialList from './MaterialList';
import translations from './translations.intl';

const propTypes = {
  folderId: PropTypes.number.isRequired,
  materials: typeMaterial,
};

class MaterialUploader extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      materials: props.materials,
      uploadingMaterials: [],
    };
  }

  onMaterialDelete = (id, name) => {
    this.setState((state) => {
      // Update UI to show the loader.
      const updatedMaterials = state.materials.map((m) => {
        if (m.id === id) {
          return { ...m, deleting: true };
        }
        return m;
      });

      return { materials: updatedMaterials };
    });

    CourseAPI.materials
      .destroy(this.props.folderId, id)
      .then(() => {
        this.setState((state) => {
          // Remove material from the list
          const materials = state.materials.filter((m) => m.id !== id);
          const successMessage = (
            <FormattedMessage
              {...{ ...translations.deleteSuccess, values: { name } }}
            />
          );

          return { materials, notification: { message: successMessage } };
        });
      })
      .catch(() => {
        this.setState((state) => {
          // Display failure message and restore the material to not deleting state
          const materials = state.materials.map((m) => {
            if (m.id === id) {
              return { ...m, deleting: false };
            }
            return m;
          });
          const failureMessage = (
            <FormattedMessage
              {...{ ...translations.deleteFail, values: { name } }}
            />
          );

          return { materials, notification: { message: failureMessage } };
        });
      });
  };

  onFileInputChange = (e) => {
    e.preventDefault();
    const fileInput = e.target;
    const { folderId } = this.props;
    const files = fileInput.files;

    const materials = [];
    for (let i = 0; i < files.length; i += 1) {
      materials.push({ name: files[i].name });
    }
    this.setState((state) => ({
      uploadingMaterials: state.uploadingMaterials.concat(materials),
    }));

    CourseAPI.materialFolders
      .upload(folderId, files)
      .then((response) => {
        this.updateMaterials(materials, response);
      })
      .catch((error) => {
        this.removeUploads(materials, error.response);
        // Set the value to null so that the files can be selected again.
        fileInput.value = null;
      });
  };

  // Remove materials from uploading list and add new materials from server reponse to existing
  // materials list.
  updateMaterials(materials, response) {
    const uploadingMaterials = this.state.uploadingMaterials.filter(
      (m) => materials.indexOf(m) === -1,
    );
    const newState = {
      uploadingMaterials,
    };

    const newMaterials = response && response.data && response.data.materials;
    if (newMaterials) {
      newState.materials = this.state.materials.concat(newMaterials);
    }
    this.setState(newState);
  }

  // Remove given materials from uploading list and display error message.
  removeUploads(materials, response) {
    const messageFromServer =
      response && response.data && response.data.message;
    const failureMessage = <FormattedMessage {...translations.uploadFail} />;
    this.setState((state) => ({
      uploadingMaterials: state.uploadingMaterials.filter(
        (m) => materials.indexOf(m) === -1,
      ),
      notification: { message: messageFromServer || failureMessage },
    }));
  }

  render() {
    return (
      <MaterialList
        materials={this.state.materials}
        uploadingMaterials={this.state.uploadingMaterials}
        notification={this.state.notification}
        onMaterialDelete={this.onMaterialDelete}
        onFileInputChange={this.onFileInputChange}
      />
    );
  }
}

MaterialUploader.propTypes = propTypes;

export default MaterialUploader;
