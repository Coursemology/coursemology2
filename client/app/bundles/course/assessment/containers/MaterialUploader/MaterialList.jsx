import PropTypes from 'prop-types';
import { FormattedMessage, defineMessages } from 'react-intl';
import { Button, Divider, List, ListSubheader } from '@material-ui/core';
import Add from '@material-ui/icons/Add';
import NotificationBar, {
  notificationShape,
} from 'lib/components/NotificationBar';
import ReactTooltip from 'react-tooltip';
import Material from './Material';

const translations = defineMessages({
  addFiles: {
    id: 'course.material.addFiles',
    defaultMessage: 'Add Files',
  },
  disableNewFile: {
    id: 'course.material.disableNewFile',
    defaultMessage:
      'This action is unavailable as the Materials Component is disabled in the Admin Settings',
  },
});

const propTypes = {
  materials: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      updated_at: PropTypes.string,
      deleting: PropTypes.bool,
    }),
  ),
  // The popup notification message
  notification: notificationShape,
  onMaterialDelete: PropTypes.func.isRequired,
  // The materials that are being uploading.
  uploadingMaterials: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      uploading: PropTypes.bool,
    }),
  ),
  onFileInputChange: PropTypes.func,
  enableMaterialsAction: PropTypes.bool,
};

const defaultProps = {
  materials: [],
  uploadingMaterials: [],
};

const styles = {
  newFileButton: {
    verticalAlign: 'middle',
  },
  uploadInput: {
    cursor: 'pointer',
    position: 'absolute',
    top: 0,
    bottom: 0,
    right: 0,
    left: 0,
    opacity: 0,
  },
};

const MaterialList = (props) => {
  const {
    materials,
    uploadingMaterials,
    onMaterialDelete,
    onFileInputChange,
    enableMaterialsAction,
  } = props;
  const header = (
    <FormattedMessage
      id="course.assessment.MaterialList.uploadedFiles"
      defaultMessage="Files"
    />
  );

  const materialNodes = materials.map((material) => (
    <Material
      key={material.id}
      id={material.id}
      name={material.name}
      updatedAt={material.updated_at}
      deleting={material.deleting}
      onMaterialDelete={onMaterialDelete}
      disabled={!enableMaterialsAction}
    />
  ));

  const uploadingMaterialNodes = uploadingMaterials.map((material) => (
    <Material
      key={material.name}
      name={material.name}
      uploading
      disabled={!enableMaterialsAction}
    />
  ));

  const newFileButton = (
    <>
      <div
        data-tip
        data-for="add-files-button"
        data-tip-disable={enableMaterialsAction}
      >
        <Button
          component="label"
          disabled={!enableMaterialsAction}
          fullWidth
          startIcon={<Add />}
          style={styles.newFileButton}
        >
          <FormattedMessage {...translations.addFiles} />
          <input
            type="file"
            multiple
            style={styles.uploadInput}
            onChange={onFileInputChange}
            disabled={!enableMaterialsAction}
          />
        </Button>
      </div>
      <ReactTooltip id="add-files-button">
        <FormattedMessage {...translations.disableNewFile} />
      </ReactTooltip>
    </>
  );

  return (
    <>
      <Divider />
      <List>
        {(materials.length > 0 || uploadingMaterials.length > 0) && (
          <ListSubheader disableSticky>{header}</ListSubheader>
        )}
        {materialNodes}
        {uploadingMaterialNodes}
        {newFileButton}
      </List>
      <Divider />
      <NotificationBar
        notification={props.notification}
        autoHideDuration={5000}
      />
    </>
  );
};

MaterialList.propTypes = propTypes;
MaterialList.defaultProps = defaultProps;

export default MaterialList;
