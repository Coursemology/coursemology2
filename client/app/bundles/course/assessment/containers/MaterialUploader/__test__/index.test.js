import { mount, shallow } from 'enzyme';
import MockAdapter from 'axios-mock-adapter';
import ProviderWrapper from 'lib/components/ProviderWrapper';
import CourseAPI from 'api/course';
import MaterialUploader from '../index';
import MaterialList from '../MaterialList';
import Material from '../Material';

const folderId = 1;
const uploadedMaterial = {
  id: 10,
  name: 'Uploaded Material',
  updated_at: '2017-01-01T08:00:00.0000000Z',
  deleting: false,
};
const materials = [1, 2].map((id) => ({
  id,
  name: `Material ${id}`,
  updated_at: `2017-01-01T0${id}:00:00.0000000Z`,
  deleting: false,
}));

// Mock axios
const client = CourseAPI.materialFolders.getClient();
const mock = new MockAdapter(client);

beforeEach(() => {
  mock.reset();
});

describe('<MaterialList />', () => {
  it('renders the component with materials', () => {
    const materialList = shallow(
      <MaterialList materials={materials} onMaterialDelete={jest.fn()} />,
    );

    expect(materialList).toMatchSnapshot();
  });
});

describe('<Material />', () => {
  it('renders the material', () => {
    const material = shallow(
      <Material
        updatedAt={uploadedMaterial.updated_at}
        {...uploadedMaterial}
      />,
    );

    expect(material).toMatchSnapshot();
  });
});

describe('<MaterialUploader />', () => {
  it('uploads the material', async () => {
    mock
      .onPut(
        `/courses/${courseId}/materials/folders/${folderId}/upload_materials`,
      )
      .reply(200, {
        materials: [uploadedMaterial],
      });

    const materialUploader = mount(
      <ProviderWrapper>
        <MaterialUploader materials={materials} folderId={folderId} />
      </ProviderWrapper>,
    );

    expect(materialUploader.find('Material')).toHaveLength(2);

    const spyUpload = jest.spyOn(CourseAPI.materialFolders, 'upload');
    // Upload a file
    materialUploader.find('input[type="file"]').simulate('change', {
      target: {
        files: [{ name: 'Uploading file' }],
      },
    });

    await sleep(1);
    expect(spyUpload).toHaveBeenCalled();
    expect(materialUploader.find('Material')).toHaveLength(3);
  });
});
