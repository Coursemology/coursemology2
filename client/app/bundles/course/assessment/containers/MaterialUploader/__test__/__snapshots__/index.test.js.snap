// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Material /> renders the material 1`] = `
<WithStyles(ListItem)
  button={true}
>
  <WithStyles(ListItemAvatar)>
    <WithStyles(Avatar)
      style={
        Object {
          "height": "32px",
          "width": "32px",
        }
      }
    >
      <pure(AssignmentIcon) />
    </WithStyles(Avatar)>
  </WithStyles(ListItemAvatar)>
  <WithStyles(ListItemText)
    primary="Uploaded Material"
    secondary={
      <div
        style={
          Object {
            "fontSize": 12,
          }
        }
      >
        01 Jan 2017, 4:00pm
      </div>
    }
  />
  <WithStyles(ListItemSecondaryAction)>
    <WithStyles(IconButton)
      onClick={[Function]}
      style={
        Object {
          "color": "black",
          "height": 24,
          "marginRight": 16,
          "padding": 4,
          "width": 24,
        }
      }
    >
      <pure(DeleteIcon)
        data-for="delete-file-button"
        data-tip={true}
        data-tip-disable={true}
      />
      <ReactTooltip
        clickable={false}
        id="delete-file-button"
        insecure={true}
        resizeHide={true}
        wrapper="div"
      >
        <FormattedMessage
          defaultMessage="This action is unavailable as the Materials Component is disabled in the Admin Settings"
          id="course.material.disableDelete"
          values={Object {}}
        />
      </ReactTooltip>
    </WithStyles(IconButton)>
  </WithStyles(ListItemSecondaryAction)>
</WithStyles(ListItem)>
`;

exports[`<MaterialList /> renders the component with materials 1`] = `
<Fragment>
  <WithStyles(Divider) />
  <WithStyles(List)>
    <WithStyles(ListSubheader)
      disableSticky={true}
    >
      <FormattedMessage
        defaultMessage="Files"
        id="course.assessment.MaterialList.uploadedFiles"
        values={Object {}}
      />
    </WithStyles(ListSubheader)>
    <Material
      deleting={false}
      disabled={true}
      id={1}
      key="1"
      name="Material 1"
      onMaterialDelete={[MockFunction]}
      updatedAt="2017-01-01T01:00:00.0000000Z"
    />
    <Material
      deleting={false}
      disabled={true}
      id={2}
      key="2"
      name="Material 2"
      onMaterialDelete={[MockFunction]}
      updatedAt="2017-01-01T02:00:00.0000000Z"
    />
    <div
      data-for="add-files-button"
      data-tip={true}
    >
      <WithStyles(Button)
        component="label"
        disabled={true}
        fullWidth={true}
        style={
          Object {
            "verticalAlign": "middle",
          }
        }
      >
        <pure(AddIcon) />
        Add Files
        <input
          disabled={true}
          multiple={true}
          style={
            Object {
              "bottom": 0,
              "cursor": "pointer",
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          type="file"
        />
      </WithStyles(Button)>
    </div>
    <ReactTooltip
      clickable={false}
      id="add-files-button"
      insecure={true}
      resizeHide={true}
      wrapper="div"
    >
      <FormattedMessage
        defaultMessage="This action is unavailable as the Materials Component is disabled in the Admin Settings"
        id="course.material.disableNewFile"
        values={Object {}}
      />
    </ReactTooltip>
  </WithStyles(List)>
  <WithStyles(Divider) />
  <Memo(NotificationBar)
    autoHideDuration={5000}
  />
</Fragment>
`;
